README.md

```diff
@@ -87,19 +87,65 @@ utility for Vault that provides higher-level abstractions like
 tree-based listing, secret generation, secure terminal password
 entry, etc.
 
+## Configuration
 
-High Availability Concerns
---------------------------
+### Template Strings
+As the base manifest shows, a full HCL configuration can be assigned to the `vault.config` property. If you're using 
+Vault in HA mode (which is recommended) you'll probably need to set vaules like `redirect_address` and `cluster_address`.
+The `vault.config` property supports the following template strings to make setting these values easier: 
 
-If you put important things in your Vault, you want it to be
-available, so you can get those important things back out again.
+**`(ip)`**
 
-Enter High Availability.
+During deployment this value will be replaced with the IP address of the instance. This will not include the protocol or
+any port information. For example for an IP based configuration:
 
-The easiest way to do high availability is to run 3 or more nodes,
-and use the Consul storage.
+```hcl
+storage "consul"{
+  path = "vault/"
+  check_timeout = "5s"
+  max_parallel = "128"
+  cluster_address = "https://(ip):8200"
+  redirect_address = "https://(ip):8200"
+}
+```
+
+**`(index)`**
+
+During deployment this value will be replaced with the index of the instance. This can be particularly useful for DNS
+configuration values. For example, if you were deploying 3 instances, this would ensure each one had a unique DNS value
+in its configuration:
+
+```hcl
+storage "consul"{
+  path = "vault/"
+  check_timeout = "5s"
+  max_parallel = "128"
+  cluster_address = "https://prod-vault-(index).yoursite.biz"
+  redirect_address = "https://prod-vault-(index).yoursite.biz"
+}
+```
+
+### Certificate Management
+
+Your Vault configuration is likely going to require TLS certificates on the instance. This Bosh release's `tls` property can
+be used to provide these certificates:
+
+```yaml
+tls:
+  - name: "my_tls_cert"
+    cert: ((tls_certificate_content))
+    key: ((tls_key_content))
+  - name: "other_tls_cert"
+    cert: ((other_tls_certificate_content))
+    key: ((other_tls_key_content))
+```
```

Comment:
are these content blocks (the ones defined with `(( ))`) described somewhere in this README? if not perhaps we could use fake static content instead.

Reply:
They aren't described in the README explicitly. They are used in the manifest that's committed and hinted at in the usage section where we mention things being created. Maybe better fake content would be like: `cert: | -----BEGIN CERTIFICATE-----"`

Reply:
yeah I think that might be slightly easier to follow